types/seamless-immutable/index.d.ts

```diff
@@ -105,7 +105,7 @@ declare namespace SeamlessImmutable {
             map<TTarget>(mapFuction: (item: T) => TTarget): Immutable<TTarget[]>;
             filter(filterFunction: (item: T) => boolean): Immutable<T[]>;
             slice(start?: number, end?: number): Immutable<T[]>;
-            concat(...arr: T[]): Immutable<T[]>;
+            concat(...arr: Array<T[]|T>): Immutable<T[]>;
```

Comment:
It might be clearer to present this as 2 separate overloads:
```typescript
concat(...arr: T[]): Immutable<T[]>;
concat(arr: T[]): Immutable<T[]>;
```

Reply:
Split into overloads

Reply:
Although this is inaccurate since it doesn't allow for mixed arguments, which the `concat` does function does.

Reply:
I didn't know that about concat, I just assume it only took one or the other. Ignore my suggestion!